src/test/java/com/google/api/generator/engine/ast/StringObjectValueTest.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 Google LLC
```

Comment:
Tests cases look good, and (as before), let's add the special characters from the baseline file to this test. Some questions to consider:
- The [baseline file](https://github.com/googleapis/gapic-generator/blob/master/src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline) currently doesn't handle certain special characters appropriately, and they turn up under different strings. Can we verify that we'll be handling them properly in the microgenerator? 
- What are other types of special characters in the same "family" of the ones that are not handled well in the monolith today?
- Do proto comments have other special characters? Doing a cursory regex search in third_party as a sanity check, and including any special characters found, should be sufficient.

Reply:
talked offline w/Mira:

1. there will be one escaper that would take care of all special characters (including those in comment or code), and we will call it from stringObjectValue class and comment classes to help escape special characters.
2. Currently the monolith does not handle the special characters in comment like [this](https://github.com/googleapis/gapic-generator/blob/master/src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline#L2574). The escaper will be called from Comment class to manage that (in a separate PR).

Reply:
Adding points after implementation
3. Created JavaEscaperClass to extend Guava Escaper which enables us to override escape() method and call it in static context. The escape() can throw exception of nullPointerException & IllegalArgumentException, we handle them at one place and throw as RuntimeException with messages.
4. build escaper manually with replacement of special characters.
5. After looking into third party proto files, we have `&nbsp; &#40` etc, which should stay the same when building string object value. Also added unit tests for US punctions(`#$%&'()*+,-./:....`), special characters in comment `<>&"@` etc. Unicode is not included since there's no usage in the current proto file set.
@miraleung Thanks!